services/workflows-service/src/data-analytics/data-analytics.service.ts

```diff
@@ -471,43 +472,57 @@ export class DataAnalyticsService {
     ) / 59);
       `,
     );
+  }
+
+  async evaluateMultipleMerchantsOneCounterparty({
```

Comment:
1. The rule refers to a function called `evaluateTransactionAvg`
2. Please don't make a generic function. Generic functions are very much not readable and hard to maintain. Try to make it specific to this rule.